File: nodejs/device/raspberrypi/examples/mqtt-subscribe.js
/**
 *   This sample code demo receive value from QIoT Suite Lite by MQTT protocol
 *   requirement:
 *   -- npm install
 *   run command: node mqtt-subscribe.js
 */

var qiot = require('./lib/qiot');

/**
 * Setup connection options
 */
var connection = new qiot(qiot.protocol.MQTT);
var connection_option = connection.readResource('./res/resourceinfo.json', './ssl/');
connection.connect(connection_option);
// subscribe when client ready
connection.on('connect', function(data) {
    // TODO: you could replace "temp" by any resource id set form QIoT Suite Lite
    connection.subscribeById("temp");
});

/**
 * Receive data of QIoT Suite Lite.
 */
connection.on('message', function(data) {
    switch (data.id) {
        case "temp":
            console.log("temp : " + data.message.value);
            console.log("------------------------")
            break;
        default:
            break;
    }
});

File: python/device/mtk-linkit-7688-duo/examples/mqtt_subscribe.py
import os
import random
import time
import json
from lib import qiot

"""
	This sample code demo receive value from QIoT Suite Lite by MQTT(S) protocol  
	requirement:
	-- opkg update
	-- opkg install distribute
	-- opkg install python-openssl
	-- easy_install pip
	-- pip install paho-mqtt
	run command: python mqtt_subscribe.py
"""

"""
	Setup connection options
"""
connection = None
connection = qiot.connection(qiot.protocol.MQTT)
connection_options = connection.read_resource('./res/resourceinfo.json', '/ssl/')

"""
	Receive data of QIoT Suite Lite.
"""
def on_message(event_trigger,data):
	message =json.loads(data["message"].payload)
	if(data['id']=='temp'):
		print "temp : " + str(message['value'])
    	print "------------------------"

def on_connect(event_trigger,data):
	print "client ready"
	connection.subscribe_by_id("temp")


connection.on("connect",on_connect)
connection.on("message",on_message)
connection.connect(connection_options)

while 1:
	pass